agent/engine/docker_task_engine.go

```diff
@@ -611,8 +604,16 @@ func (engine *DockerTaskEngine) StateChangeEvents() chan statechange.Event {
 }
 
 // AddTask starts tracking a task
+// TODO: no error is returned from this method. remove the return type
 func (engine *DockerTaskEngine) AddTask(task *api.Task) error {
-	task.PostUnmarshalTask(engine.cfg, engine.credentialsManager)
+	err := task.PostUnmarshalTask(engine.cfg, engine.credentialsManager, engine.resourceFields)
+	if err != nil {
+		seelog.Errorf("Task engine [%s]: unable to add task to the engine: %v", task.Arn, err)
+		task.SetKnownStatus(api.TaskStopped)
+		task.SetDesiredStatus(api.TaskStopped)
+		engine.emitTaskEvent(task, err.Error())
+		return nil
```

Comment:
Please modify this as per https://github.com/aws/amazon-ecs-agent/pull/1357#discussion_r184716456